Detectors/TPC/simulation/include/TPCSimulation/SpaceCharge.h

```diff
@@ -0,0 +1,181 @@
+  static constexpr float RadiusOuter = 245.;    //! outer radius of the TPC active area
+
+  const int mInterpolationOrder; ///< order for interpolation of lookup tables: 2==quadratic, >2==cubic spline
+
+  const int mNZSlices;          ///< number of z slices used in lookup tables
+  const int mNPhiBins;          ///< number of phi bins used in lookup tables
+  const int mNRBins;            ///< number of r bins used in lookup tables
+  const float mLengthZSlice;    ///< length of one z bin (cm)
+  const float mLengthTimeSlice; ///< ion drift time for one z slice (us)
+  const float mWidthPhiBin;     ///< width of one phi bin (radians)
+  const float mLengthRBin;      ///< length of one r bin (cm)
+
+  std::vector<double> mCoordZ;   ///< vector wiht coodinates of the z bins
+  std::vector<double> mCoordPhi; ///< vector wiht coodinates of the phi bins
+  std::vector<double> mCoordR;   ///< vector wiht coodinates of the r bins
+
+  bool mUseInitialSCDensity;          ///< Flag for the use of an initial space-charge density at the beginning of the simulation
+  bool mInitLookUpTables;             ///< Flag to indicate if lookup tables have been calculated
+  float mTimeInit;                    ///< time of last update of lookup tables
+  SCDistortionType mSCDistortionType; ///< Type of space-charge distortions
+
+  AliTPCSpaceCharge3DCalc mLookUpTableCalculator; ///< object to calculate and store correction and distortion lookup tables
+
+  /// TODO: check fastest way to order std::vectors
+  std::vector<std::vector<float>> mSpaceChargeDensityA; ///< space-charge density on the A side, stored in C/m^3 (z)(phi*r), ordering: z=[0,250], ir+iphi*nRBins
+  std::vector<std::vector<float>> mSpaceChargeDensityC; ///< space-charge density on the C side, stored in C/m^3 (z)(phi*r), ordering: z=[0,-250], ir+iphi*nRBins
+
+  TMatrixD** mMatrixLocalIonDriftDzA;                      ///< matrix to store local ion drift in z direction along E field on A side
+  TMatrixD** mMatrixLocalIonDriftDzC;                      ///< matrix to store local ion drift in z direction along E field on A side
+  TMatrixD** mMatrixLocalIonDriftDrphiA;                   ///< matrix to store local ion drift in rphi direction along E field on A side
+  TMatrixD** mMatrixLocalIonDriftDrphiC;                   ///< matrix to store local ion drift in rphi direction along E field on A side
+  TMatrixD** mMatrixLocalIonDriftDrA;                      ///< matrix to store local ion drift in radial direction along E field on A side
+  TMatrixD** mMatrixLocalIonDriftDrC;                      ///< matrix to store local ion drift in radial direction along E field on C side
+  AliTPCLookUpTable3DInterpolatorD* mLookUpLocalIonDriftA; ///< lookup table for local ion drift along E field on A side
+  AliTPCLookUpTable3DInterpolatorD* mLookUpLocalIonDriftC; ///< lookup table for local ion drift along E field on C side
```

Comment:
all of these should be expressed as `unique_ptr`s. I also cannot find where those objects are deleted, which would of course not be needed in case of `unique_ptr`.

Reply:
The ownership of the TMatrixDs is moved to AliTPCLookUpTable3DInterpolatorD, where they are also deleted.
For a future version, the need of the TMatrixDs can be eliminated by introducing getters for those and adding a new constructor to AliTPCLookUpTable3DInterpolatorD. This requires modifications of AliTPCCommon and a new tag.
So for the moment, I will only change the AliTPCLookUpTable3DInterpolatorD pointers to unique_ptrs.